content/git/encyclopedia/what-is-push-in-git.md

```diff
@@ -0,0 +1,75 @@
+---
+Title: "What is Push in Git?"
+Subjects:
+  - "Bash/Shell"
+  - "Developer Tools"
+Tags: 
+  - "Git"
+  - "GitHub"
+Catalog Content:
+  - "https://www.codecademy.com/learn/learn-git"
+  - "https://www.codecademy.com/learn/learn-the-command-line"
+---
+
+In Git, pushing is the process of moving code from one repository to another one. Often this is used to move code from a local machine to a remote one, or to a repository hosting service like GitHub.
+
+Pushing a repository allows a programmer to maintain an easily accessible backup in a second location. It is also used when collaborating with people that would not have access to the computer that the code is written on.
+
+## How do I push?
+
+Prerequisites to pushing from within a repository include the following:
+
+* Being inside of a repository (if you're not, the next step will give you an error)
+* Having a clean `git status` (no uncommitted, but tracked, files):
+
+    ```
+    $ git status
+
+    On branch main
+    nothing to commit, working tree clean
+    ```
+
+* Having a valid remote repository set, in our case there is remote repository called companyname-dev:
+
+    ```
+    $ git remote -v
+
+    companyname-dev	https://github.com/CompanyName/product-dev.git (fetch)
+    companyname-dev	https://github.com/CompanyName/product-dev.git (push)
+    ```
+
+Once you are at this stage, pushing is as simple as running the following command from the command line:
+
+```
+$ git push <remote-name> <branch-name>
```

Comment:
I would move the `$ git push <remote-name> <branch-name>` up because that's the main syntax and we want people to find what they are looking for fast.